---
title: "Incorporator rRNA operon copy number"
author: "Samuel Barnett"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  github_document:
    toc: true
    toc_depth: 2
---

## Introduction

We are generally interested in the life history of the bacteria active in soil C cycling in our soils. To get at this a bit we will look at the predicted rRNA operon copy number, which tends to correlate with maximum growth rate, and see if these values vary across the incorporators.

### Initialization
```{r, message=FALSE, warning=FALSE}
# For data handling
library(dplyr)
library(phyloseq)

# For plotting
library(ggplot2)

# Set color schemes
eco.col = c(agriculture="#00BA38", meadow="#619CFF", forest="#F8766D")
landuse.col = c("Cropland"="#00BA38", "Old-field"="#619CFF", "Forest"="#F8766D")

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

# Dataframe for converting day to a time period (early and late)
subtime.df = data.frame(day = c(1, 3, 6, 14, 1, 3, 6, 14,
                                6, 14, 30, 6, 14, 30, 6, 14, 30),
                        substrate = c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                      "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                      "13C-Van", "13C-Van", "13C-Van", 
                                      "13C-Cel", "13C-Cel", "13C-Cel",
                                      "13C-Pal", "13C-Pal", "13C-Pal"),
                        period = c("Early", "Early", "Late", "Late",
                                   "Early", "Early", "Late", "Late",
                                   "Early", "Late", "Late",
                                   "Early", "Early", "Late",
                                   "Early", "Early", "Late"),
                        Ctype = c("DOM", "DOM", "DOM", "DOM",
                                  "DOM", "DOM", "DOM", "DOM",
                                  #"Van", "Van", "Van",
                                  "DOM", "DOM", "DOM",
                                  "POM", "POM", "POM",
                                  "POM", "POM", "POM"))

```

### Data
```{r, message=FALSE, warning=FALSE}
# Get the l2fc dataframe and add in columns indicating the land-use, labeled substrate, and day
l2fc.df = readRDS(file = "/Users/sambarnett/Documents/Buckley Lab/FullCyc2/fullcyc2_l2fc_testoutput.rds") %>%
  mutate(ecosystem  = factor(gsub(".+ecosystem == [ \']*([A-z]+).+", "\\1", .id),
                             levels = c("agriculture", "meadow", "forest")),
         day  = gsub(".+day == [ \']*([0-9]+).+", "\\1", .id),
         substrate = factor(gsub(".+(13C-[A-z]+).+", "\\1", .id),
                            levels = c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))

# Mapping OTUs to FullCyc2 (Both BLAST and mothur aligned)
BLAST2FC1.df = read.table("/Users/sambarnett/Documents/Buckley Lab/FullCyc2/FC2_to_FC1_blast.txt", header=FALSE, sep="\t")
colnames(BLAST2FC1.df) = c("qaccver", "saccver", "pident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")
mothur_aligned.df = read.table("/Users/sambarnett/Documents/Buckley Lab/FullCyc2/all_combined_map.txt", header=TRUE, sep="\t")
FC1_LHs.df = read.table("/Users/sambarnett/Documents/Buckley Lab/FullCyc2/FullCyc_incorporators.txt", header=TRUE, sep="\t") %>%
  rename(FC1_OTU = OTU)

# rRNA gene copy number predictions (from paprica)
#ribosome_counts.df = read.csv("/Users/sambarnett/Desktop/FullCyc2_incorporators.bacteria.unique_seqs.csv") %>%
ribosome_counts.df = read.csv("/Users/sambarnett/Documents/Buckley Lab/FullCyc2/paprica_final/final_incorp.bacteria.unique_seqs.csv") %>%
  select(name, abundance_corrected) %>%
  mutate(count = 1/abundance_corrected) %>%
  rename(OTU = name)
```

## rRNA copy number predictions

Here lets see how the rrn (rRNA operon copy number) compares across incorporators of C from each substrate and land-use. We will do this separately for early and late C incorporators.

### Early timepoints
Early incorporators are those that assimilate C before peak mineralization.


First, lets look across substrate within land use.
```{r, fig.height=3.5, fig.width=7, message=FALSE, warning=FALSE}
sub.changes = data.frame(substrate = c("13C-Ami", "13C-Xyl", "13C-Van", "13C-Cel", "13C-Pal"), 
                         sub_long = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))
eco.changes = data.frame(ecosystem = c("agriculture", "meadow", "forest"), 
                         land_use = c("Cropland", "Old-field", "Forest"))

early_rrn.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  mutate(day = as.numeric(day)) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(period == "Early") %>%
  left_join(sub.changes, by="substrate") %>%
  left_join(eco.changes, by="ecosystem") %>%
  select(OTU, land_use, sub_long, period, Ctype) %>%
  unique() %>%
  left_join(ribosome_counts.df, by="OTU") %>%
  mutate(sub_long = factor(sub_long, levels = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))

early_substrate_rrn.kruskal.res = data.frame()
for (lu in c("Cropland", "Old-field", "Forest")){
  ktest.res = kruskal.test(count~sub_long, data=filter(early_rrn.df, land_use == lu))
  if (p.adjust(ktest.res$p.value, n=3, method="BH") < 0.05){
    posthoc.res = FSA::dunnTest(count~sub_long, data=filter(early_rrn.df, land_use == lu))
    posthoc.res = posthoc.res$res
    posthoc.cld = rcompanion::cldList(comparison = posthoc.res$Comparison,
                                      p.value = posthoc.res$P.adj,
                                      threshold = 0.05)
  } else{
    posthoc.cld = data.frame(Group = NA, Letter = NA)
  }
  sub.posthoc.df = data.frame(land_use = lu, X2 = ktest.res$statistic, Pvalue = ktest.res$p.value,
                              padj = p.adjust(ktest.res$p.value, n=3, method="BH"),
                              sub_long = posthoc.cld$Group, group = posthoc.cld$Letter,
                              stringsAsFactors = FALSE) %>%
    mutate(sub_long = gsub("acid", " acid", sub_long))
  early_substrate_rrn.kruskal.res = rbind(early_substrate_rrn.kruskal.res, sub.posthoc.df)
}
early_substrate_rrn.kruskal.res = early_substrate_rrn.kruskal.res %>%
  mutate(sub_long = factor(sub_long, levels = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))

early_substrate_rrn.plot = ggplot(data=early_rrn.df, aes(x=sub_long, y=count)) +
  geom_violin(aes(fill=sub_long)) +
  geom_boxplot(width=0.15) +
  geom_text(data=early_substrate_rrn.kruskal.res, 
            y=max(early_rrn.df$count, na.rm = TRUE)+1, 
            aes(x=sub_long, label=group), size=4) +
  lims(y=c(0, max(early_rrn.df$count, na.rm = TRUE)+1)) +
  labs(x="Substrate", y="Predicted rRNA copy number") +
  facet_wrap(~land_use, nrow=1) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1),
        text = element_text(size=12),
        legend.position = "none")

early_substrate_rrn.plot

early_substrate_rrn.kruskal.res

```

Next lets look across land use within substrate.
```{r, fig.height=3.5, fig.width=7, message=FALSE, warning=FALSE}
early_eco_rrn.kruskal.res = data.frame()
for (subC in c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid")){
  ktest.res = kruskal.test(count~land_use, data=filter(early_rrn.df, sub_long == subC))
  if (p.adjust(ktest.res$p.value, n=5, method="BH") < 0.05){
    posthoc.res = FSA::dunnTest(count~land_use, data=mutate(filter(early_rrn.df, sub_long == subC), land_use = gsub("-", "", land_use)))
    posthoc.res = posthoc.res$res
    posthoc.cld = rcompanion::cldList(comparison = posthoc.res$Comparison,
                                      p.value = posthoc.res$P.adj,
                                      threshold = 0.05)
  } else{
    posthoc.cld = data.frame(Group = NA, Letter = NA)
  }
  sub.posthoc.df = data.frame(sub_long = subC, X2 = ktest.res$statistic, Pvalue = ktest.res$p.value,
                              padj = p.adjust(ktest.res$p.value, n=5, method="BH"),
                              land_use = posthoc.cld$Group, group = posthoc.cld$Letter,
                              stringsAsFactors = FALSE) %>%
    mutate(land_use = gsub("Oldfield", "Old-field", land_use))
  early_eco_rrn.kruskal.res = rbind(early_eco_rrn.kruskal.res, sub.posthoc.df)
}
early_eco_rrn.kruskal.res = early_eco_rrn.kruskal.res %>%
  mutate(sub_long = factor(sub_long, levels = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))


early_eco_rrn.plot = ggplot(data=early_rrn.df, aes(x=land_use, y=count)) +
  geom_violin(aes(fill=land_use)) +
  geom_boxplot(width=0.15) +
  geom_text(data=early_eco_rrn.kruskal.res, 
            y=max(early_rrn.df$count, na.rm = TRUE)+1, 
            aes(x=land_use, label=group), size=4) +
  scale_fill_manual(values=landuse.col) +
  lims(y=c(0, max(early_rrn.df$count, na.rm = TRUE)+1)) +
  labs(x="Land use", y="Predicted rRNA gene copy number") +
  facet_wrap(~sub_long, nrow=1) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1),
        text = element_text(size=12),
        legend.position = "none")

early_eco_rrn.plot
early_eco_rrn.kruskal.res


#ggsave(early_eco_rrn.plot, filename = "/Users/sambarnett/Documents/Dissertation/figures/fig2_4.tiff", 
#       device = "tiff", width = 7, height = 3.5, units = "in")


```

### Late timepoints

Late incorporators are those that assimilate C after peak mineralization.

First look across substrate within land use.
```{r, fig.height=3.5, fig.width=7, message=FALSE, warning=FALSE}
sub.changes = data.frame(substrate = c("13C-Ami", "13C-Xyl", "13C-Van", "13C-Cel", "13C-Pal"), 
                         sub_long = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))
eco.changes = data.frame(ecosystem = c("agriculture", "meadow", "forest"), 
                         land_use = c("Cropland", "Old-field", "Forest"))

late_rrn.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  mutate(day = as.numeric(day)) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(period == "Late") %>%
  left_join(sub.changes, by="substrate") %>%
  left_join(eco.changes, by="ecosystem") %>%
  select(OTU, land_use, sub_long, period, Ctype) %>%
  unique() %>%
  left_join(ribosome_counts.df, by="OTU") %>%
  mutate(sub_long = factor(sub_long, levels = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))

late_substrate_rrn.kruskal.res = data.frame()
for (lu in c("Cropland", "Old-field", "Forest")){
  ktest.res = kruskal.test(count~sub_long, data=filter(late_rrn.df, land_use == lu))
  if (p.adjust(ktest.res$p.value, n=3, method="BH") < 0.05){
    posthoc.res = FSA::dunnTest(count~sub_long, data=filter(late_rrn.df, land_use == lu))
    posthoc.res = posthoc.res$res
    posthoc.cld = rcompanion::cldList(comparison = posthoc.res$Comparison,
                                      p.value = posthoc.res$P.adj,
                                      threshold = 0.05)
  } else{
    posthoc.cld = data.frame(Group = NA, Letter = NA)
  }
  sub.posthoc.df = data.frame(land_use = lu, X2 = ktest.res$statistic, Pvalue = ktest.res$p.value,
                              padj = p.adjust(ktest.res$p.value, n=3, method="BH"),
                              sub_long = posthoc.cld$Group, group = posthoc.cld$Letter,
                              stringsAsFactors = FALSE) %>%
    mutate(sub_long = gsub("acid", " acid", sub_long))
  late_substrate_rrn.kruskal.res = rbind(late_substrate_rrn.kruskal.res, sub.posthoc.df)
}
late_substrate_rrn.kruskal.res = late_substrate_rrn.kruskal.res %>%
  mutate(sub_long = factor(sub_long, levels = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))

late_substrate_rrn.plot = ggplot(data=late_rrn.df, aes(x=sub_long, y=count)) +
  geom_violin(aes(fill=sub_long)) +
  geom_boxplot(width=0.15) +
  geom_text(data=filter(late_substrate_rrn.kruskal.res, padj < 0.05), 
            y=max(late_rrn.df$count, na.rm = TRUE)+1, 
            aes(x=sub_long, label=group), size=4) +
  lims(y=c(0, max(late_rrn.df$count, na.rm = TRUE)+1)) +
  labs(x="Substrate", y="Predicted rRNA copy number") +
  facet_wrap(~land_use, nrow=1) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1),
        text = element_text(size=12),
        legend.position = "none")

late_substrate_rrn.plot

late_substrate_rrn.kruskal.res
```

Next look across land-use within substrate.

```{r, fig.height=3.5, fig.width=7, message=FALSE, warning=FALSE}
late_eco_rrn.kruskal.res = data.frame()
for (subC in c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid")){
  ktest.res = kruskal.test(count~land_use, data=filter(late_rrn.df, sub_long == subC))
  if (p.adjust(ktest.res$p.value, n=5, method="BH") < 0.05){
    posthoc.res = FSA::dunnTest(count~land_use, data=mutate(filter(late_rrn.df, sub_long == subC), land_use = gsub("-", "", land_use)))
    posthoc.res = posthoc.res$res
    posthoc.cld = rcompanion::cldList(comparison = posthoc.res$Comparison,
                                      p.value = posthoc.res$P.adj,
                                      threshold = 0.05)
  } else{
    posthoc.cld = data.frame(Group = NA, Letter = NA)
  }
  sub.posthoc.df = data.frame(sub_long = subC, X2 = ktest.res$statistic, Pvalue = ktest.res$p.value,
                              padj = p.adjust(ktest.res$p.value, n=5, method="BH"),
                              land_use = posthoc.cld$Group, group = posthoc.cld$Letter,
                              stringsAsFactors = FALSE) %>%
    mutate(land_use = gsub("Oldfield", "Old-field", land_use))
  late_eco_rrn.kruskal.res = rbind(late_eco_rrn.kruskal.res, sub.posthoc.df)
}
late_eco_rrn.kruskal.res = late_eco_rrn.kruskal.res %>%
  mutate(sub_long = factor(sub_long, levels = c("Amino acids", "Xylose", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))


late_eco_rrn.plot = ggplot(data=late_rrn.df, aes(x=land_use, y=count)) +
  geom_violin(aes(fill=land_use)) +
  geom_boxplot(width=0.15) +
  geom_text(data=filter(late_eco_rrn.kruskal.res, padj < 0.05), 
            y=max(late_rrn.df$count, na.rm = TRUE)+1, 
            aes(x=land_use, label=group), size=4) +
  scale_fill_manual(values=landuse.col) +
  lims(y=c(0, max(late_rrn.df$count, na.rm = TRUE)+1)) +
  labs(x="Land-use", y="Predicted rRNA gene copy number") +
  facet_wrap(~sub_long, nrow=1) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1),
        text = element_text(size=12),
        legend.position = "none")

late_eco_rrn.plot
late_eco_rrn.kruskal.res

#ggsave(late_eco_rrn.plot, filename = "/Users/sambarnett/Documents/Dissertation/figures/figS2_10.tiff", 
#       device = "tiff", width = 7, height = 3.5, units = "in")
```

## Plot 1 for publication (Fig 5)

Here is the plot used in the manuscript.

```{r, fig.height=2, fig.width=3.46457, message=FALSE, warning=FALSE}
early_eco_rrn.kruskal.res = data.frame()
for (subC in c("Xylose", "Amino acids", "Vanillin", "Cellulose", "Palmitic acid")){
  ktest.res = kruskal.test(count~land_use, data=filter(early_rrn.df, sub_long == subC))
  if (p.adjust(ktest.res$p.value, n=5, method="BH") < 0.05){
    posthoc.res = FSA::dunnTest(count~land_use, data=mutate(filter(early_rrn.df, sub_long == subC), land_use = gsub("-", "", land_use)))
    posthoc.res = posthoc.res$res
    posthoc.cld = rcompanion::cldList(comparison = posthoc.res$Comparison,
                                      p.value = posthoc.res$P.adj,
                                      threshold = 0.05)
  } else{
    posthoc.cld = data.frame(Group = NA, Letter = NA)
  }
  sub.posthoc.df = data.frame(sub_long = subC, X2 = ktest.res$statistic, Pvalue = ktest.res$p.value,
                              padj = p.adjust(ktest.res$p.value, n=5, method="BH"),
                              land_use = posthoc.cld$Group, group = posthoc.cld$Letter,
                              stringsAsFactors = FALSE) %>%
    mutate(land_use = gsub("Oldfield", "Old-field", land_use))
  early_eco_rrn.kruskal.res = rbind(early_eco_rrn.kruskal.res, sub.posthoc.df)
}
early_eco_rrn.kruskal.res = early_eco_rrn.kruskal.res %>%
  mutate(sub_long = factor(sub_long, levels = c("Xylose", "Amino acids", "Vanillin", "Cellulose", "Palmitic acid"))) %>%
  mutate(land_use = factor(land_use, levels = c("Cropland", "Old-field", "Forest")))


early_eco_rrn.plot = ggplot(data=early_rrn.df, aes(x=land_use, y=count)) +
  geom_violin(aes(fill=land_use)) +
  geom_boxplot(width=0.15, outlier.size = 0.5) +
  geom_text(data=early_eco_rrn.kruskal.res, 
            y=max(early_rrn.df$count, na.rm = TRUE)+1, 
            aes(x=land_use, label=group), size=(5*5/14)) +
  scale_fill_manual(values=landuse.col) +
  lims(y=c(0, max(early_rrn.df$count, na.rm = TRUE)+1)) +
  labs(x="Land use", y="Predicted rRNA\ngene copy number") +
  facet_wrap(~factor(sub_long, levels=c("Xylose", "Amino acids", "Vanillin", "Cellulose", "Palmitic acid")), nrow=1) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1, size=6),
        axis.text.y = element_text(size=6),
        axis.title = element_text(size=7),
        axis.ticks = element_line(size=0.2),
        strip.text = element_text(size=6),
        legend.position = "none")

early_eco_rrn.plot


#ggsave(early_eco_rrn.plot, filename = "/Users/sambarnett/Documents/Buckley Lab/FullCyc2/manuscript/Figures/Fig5.tiff", 
#       device = "tiff", width = 3.46457, height = 2, units = "in")


```






