---
title: "Incorporators"
author: "Samuel Barnett"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    toc: true
    toc_float: true
---

## Introduction

Now that we have use MW-HR-SIP to identify incorporators, I want to take a closer look at these labeled OTUs. Specifically, I want to see if they differ systematically across land use regimes, substrates and timepoints. 

#### Initialization
```{r}
# For data handling
library(dplyr)
library(phyloseq)

# For analysis
library(nlme)
library(lsmeans)

# For plotting
library(ggplot2)
library(eulerr)
library(grid)
library(gridExtra)

# Set color schemes
eco.col = c(agriculture="#00BA38", meadow="#619CFF", forest="#F8766D")

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}
```

#### Data
Incorporator status can be found in the log2 fold change dataframe.
```{r}
# Get the l2fc dataframe and add in columns indicating the land-use, labeled substrate, and day
l2fc.df = readRDS(file = "/Users/sambarnett/Documents/Buckley Lab/FullCyc2/fullcyc2_l2fc_testoutput.rds") %>%
  mutate(ecosystem  = factor(gsub(".+ecosystem == [ \']*([A-z]+).+", "\\1", .id),
                             levels = c("agriculture", "meadow", "forest")),
         day  = gsub(".+day == [ \']*([0-9]+).+", "\\1", .id),
         substrate = factor(gsub(".+(13C-[A-z]+).+", "\\1", .id),
                            levels = c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))

```

The OTU abundances and phylogeny can be found in the master phyloseq object.

```{r}
# Import phyloseq object
physeq = readRDS("/Users/sambarnett/Documents/Buckley Lab/FullCyc2/fullcyc2_backups_8_8_19/phyloseq/fullcyc2physeq.RDS")

# Subset to just the unfractionated samples and remove the controls
unfrac.physeq = subset_samples(physeq, exp_type == "Unfractionated" & sample_type == "unknown")
bulk.physeq = subset_samples(physeq, exp_type == "bulk" & sample_type == "unknown")
physeq = NULL

# Remove samples MR.F.12C-Con.D0.R2 and MR.M.12C-Con.D0.R2 since these were likely accidently switched. Their correct, resequenced versions are called MR.F.12C-Con.D0.R2_primer2 and MR.M.12C-Con.D0.R2_primer2
unfrac.physeq = subset_samples(unfrac.physeq, !(X.Sample %in% c("MR.F.12C-Con.D0.R2", "MR.M.12C-Con.D0.R2")))

# Remove non-bacteria (aka Archaea)
unfrac.physeq = subset_taxa(unfrac.physeq, Domain == "Bacteria")
bulk.physeq = subset_taxa(bulk.physeq, Domain == "Bacteria")
  
# Add in a different phylogenetic tree. The one in the phyloseq might be an older version.
tree = read_tree("/Users/sambarnett/Documents/Buckley Lab/FullCyc2/fullcyc2_backups_8_8_19/fullcyc2.bacteria.cogent.tree")
phy_tree(unfrac.physeq) = tree
phy_tree(bulk.physeq) = tree

# Remove any OTUs no longer found in the samples
unfrac.physeq = prune_taxa(taxa_sums(unfrac.physeq) > 0, unfrac.physeq)
bulk.physeq = prune_taxa(taxa_sums(bulk.physeq) > 0, bulk.physeq)
```

For many of the following analyses I want a rarefied OTU table. This is one way to correct for differing sequencing depths across all my samples. I will set the seed for this process so that I can replicate this analysis if necessary (seed = 4242).

```{r}
unfrac.rare.physeq = rarefy_even_depth(unfrac.physeq, rngseed=4242)
bulk.rare.physeq = rarefy_even_depth(bulk.physeq, rngseed=4242)
```

## Initial stats

Before proceeding too far, lets just take a look a few inital stats.

```{r}
# Total number of OTUs passing filtering (those that went into MW-HR-SIP)
print(paste("There are", nrow(unique(select(l2fc.df, OTU))), "total OTUs passing filter"))

# Total number of incorporators
print(paste("There are", nrow(unique(select(filter(l2fc.df, padj < 0.05), OTU))), "total incorporators"))

# Total number of incorporators for each land-use
for (eco in c("agriculture", "meadow", "forest")){
  print(paste("There are", nrow(unique(select(filter(l2fc.df, padj < 0.05 & ecosystem == eco), OTU))), "incorporators from", eco))
}

# Total number of incorporators for each substrate
for (sub in c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")){
  print(paste("There are", nrow(unique(select(filter(l2fc.df, padj < 0.05 & substrate == sub), OTU))), "incorporators of", sub))
}

# Number of incorporator phyla
print(paste("There are", nrow(unique(select(filter(l2fc.df, padj < 0.05, !(is.na(Phylum))), Phylum))), "incorporator phyla"))


```

Now break this up into table format
```{r}
l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem, substrate) %>%
  unique %>%
  group_by(ecosystem, substrate) %>%
  summarize(n_OTU = n())
```


## Venn diagrams
The first thing I want to do is see how incorporators differ between land-use regimes. To do this I will generate venn diagrams to see how many incoporators are within each land use and how many are shared.

### Across ecosystems

First make the plot showing which land use regimes incorporators are labeled in and which ones they are at least detected in.

```{r, fig.height=2, fig.width=3.46457}
# Get a table of each OTU and its labeling or detection pattern across land use regimes.
incorp_eco.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem) %>%
  unique %>%
  mutate(value = as.logical("TRUE")) %>%
  tidyr::spread(key=ecosystem, value=value) %>%
  select(-OTU) %>%
  mutate(test = "Labeling")
PA_eco.df = l2fc.df %>%
  group_by(OTU) %>%
  mutate(min_padj = min(padj)) %>%
  ungroup %>%
  filter(min_padj < 0.05) %>%
  select(OTU, ecosystem) %>%
  unique %>%
  mutate(value = as.logical("TRUE")) %>%
  tidyr::spread(key=ecosystem, value=value) %>%
  select(-OTU) %>%
  mutate(test = "Detection")
venn_eco.df = rbind(incorp_eco.df, PA_eco.df)
venn_eco.df[is.na(venn_eco.df)] = as.logical("FALSE")

venn_eco.plot = plot(euler(venn_eco.df, by = list(test)), 
                     legend = FALSE, labels = FALSE,
                     #legend = list(labels = c("Cropland", "Old-field", "Forest"), fontsize=5),
                     quantities = list(TRUE, fontsize=5), strip=FALSE, fills=eco.col)

# Add axis labels
x.grob <- textGrob("Detection                                      Labeling", gp=gpar(fontface="bold", col="black", fontsize=7))

venn_eco.plot = grid.arrange(arrangeGrob(venn_eco.plot, top = x.grob))
venn_eco.plot


```

### Across ecosystems within substrate

First make the plot showing which land use regimes incorporators are labeled in and which ones they are at least detected in.

```{r, fig.height=3.5, fig.width=7}
# Get a table of each OTU and its labeling or detection pattern across land use regimes.
incorp_subeco.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem, substrate) %>%
  unique %>%
  mutate(value = as.logical("TRUE")) %>%
  tidyr::spread(key=ecosystem, value=value) %>%
  select(agriculture, meadow, forest, substrate) %>%
  mutate(test = "Labeling")

incorp_sub.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, substrate) %>%
  unique

PA_subeco.df = l2fc.df %>%
  group_by(OTU) %>%
  mutate(min_padj = min(padj)) %>%
  ungroup %>%
  filter(min_padj < 0.05) %>%
  select(OTU, ecosystem) %>%
  unique %>%
  mutate(value = as.logical("TRUE")) %>%
  tidyr::spread(key=ecosystem, value=value) %>%
  left_join(incorp_sub.df, by = "OTU") %>%
  select(-OTU) %>%
  mutate(test = "Detection")

venn_subeco.df = rbind(incorp_subeco.df, PA_subeco.df)
venn_subeco.df[is.na(venn_subeco.df)] = as.logical("FALSE")

# Plot the venn diagrams together
venn_subeco.plot = plot(euler(venn_subeco.df, by = list(test, substrate)), 
                        quantities = TRUE, labels = NULL,
                        fills=eco.col, strip=FALSE)

ggsave(venn_subeco.plot, file="/Users/sambarnett/Desktop/Substrate_venns_unlab.tiff", device = "tiff")

# Add axis labels
x.grob <- textGrob("Xylose       Amino acids        Vanillin        Cellulose     Palmitic acid", gp=gpar(fontface="bold", col="black", fontsize=15))
y.grob <- textGrob("Detection    Labeling", 
                   gp=gpar(fontface="bold", col="black", fontsize=15), rot=90)

grid.arrange(arrangeGrob(venn_subeco.plot, top = x.grob, left = y.grob))

```

### Across time within substrate and ecosystem

```{r, fig.height=3.34646, fig.width=3.34646}
# Dataframe for converting day to a time period (early and late)
timeperiods = data.frame(day = c(1, 3, 6, 14, 1, 3, 6, 14,
                                 6, 14, 30, 6, 14, 30, 6, 14, 30),
                         substrate = c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                       "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                       "13C-Van", "13C-Van", "13C-Van", 
                                       "13C-Cel", "13C-Cel", "13C-Cel",
                                       "13C-Pal", "13C-Pal", "13C-Pal"),
                         period = c("Early", "Early", "Late", "Late",
                                    "Early", "Early", "Late", "Late",
                                    "Early", "Late", "Late",
                                    "Early", "Early", "Late",
                                    "Early", "Early", "Late"))

# Get a table of each OTU and its labeling or detection pattern across land use regimes.
venn_period.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  mutate(day = as.numeric(day)) %>%
  left_join(timeperiods) %>%
  select(OTU, ecosystem, substrate, period) %>%
  unique %>%
  mutate(value = as.logical("TRUE")) %>%
  tidyr::spread(key=period, value=value) %>%
  select(ecosystem, substrate, Early, Late) %>%
  replace(is.na(.), as.logical("FALSE")) %>%
  mutate(ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))

# Plot the venn diagrams together
per.col = c(Early = "yellow", Late = "purple")
venn_period.plot = plot(euler(venn_period.df, by = list(substrate, ecosystem)), 
                        quantities = TRUE, labels = NULL, 
                        fills = per.col, strip = FALSE)

# Add axis labels
x.grob <- textGrob("A            M            F", gp=gpar(fontface="bold", col="black", fontsize=12))
y.grob <- textGrob("Pal        Cel       Van       Ami      Xyl    ", 
                   gp=gpar(fontface="bold", col="black", fontsize=12), rot=90)

grid.arrange(arrangeGrob(venn_period.plot, top = x.grob, left = y.grob))

```

## Incorporator taxonomy

Lets take a brief look at the taxonomy of the incorporators across land-use and time for each substrate.

### OTU counts per Phylum
First I'll look specifically at the phylum (class for proteobacteria) level and how many incorporator OTUs are classified to each phylum (class for Proteobacteria).

```{r, fig.height=6.69291, fig.width=6.69291}
# Get OTU counts for each taxa
incorp_phyla.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  mutate(taxa = ifelse(is.na(Phylum), "Unclassified bacteria",
                       ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum)))) %>%
  group_by(taxa, ecosystem, substrate, day) %>%
  summarize(OTU_count = n()) %>%
  as.data.frame %>%
  mutate(eco = factor(ifelse(ecosystem == "agriculture", "A",
                             ifelse(ecosystem == "meadow", "M",
                                    ifelse(ecosystem == "forest", "F", NA))), 
                      levels = c("A", "M", "F")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")),
         day = as.numeric(day))

# Set the taxa colors
library(scales)
taxa = sort(unique(incorp_phyla.df$taxa))
taxa.col = hue_pal()(length(taxa))
names(taxa.col) = taxa

# Plot for soluble substrates (measured on days 1, 3, 6, and 14)
sol.count.plot = ggplot(data=incorp_phyla.df[incorp_phyla.df$substrate %in% c("13C-Xyl", "13C-Ami"),],
       aes(x=eco, y=OTU_count, fill=taxa)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=taxa.col) +
  labs(x = "Land-use", y="OTU count", fill="Phylum/Class") +
  theme_bw() +
  theme(legend.position = "none",
        legend.title=element_text(size=14),
        legend.text=element_text(size=12),
        axis.title = element_blank(),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_grid(day~substrate, scales="free_y")

# Plot for insoluable substrates (measured on days 6, 14, and 30)
insol.count.plot = ggplot(data=incorp_phyla.df[incorp_phyla.df$substrate %in% c("13C-Van", "13C-Cel", "13C-Pal"),],
       aes(x=eco, y=OTU_count, fill=taxa)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=taxa.col) +
  labs(x = "Land-use", y="OTU count", fill="Phylum/Class") +
  guides(fill = guide_legend(ncol=3)) +
  theme_bw() +
  theme(legend.position = "none",
        legend.title=element_blank(),
        legend.text=element_text(size=10),
        axis.title = element_blank(),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_grid(day~substrate, scales="free_y")

# Get the legend for all substrates
leg_plot = ggplot(data=incorp_phyla.df,
       aes(x=eco, y=OTU_count, fill=taxa)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=taxa.col) +
  labs(x = "Land-use", y="OTU count", fill="Phylum/Class") +
  guides(fill = guide_legend(ncol=4)) +
  theme_bw() +
  theme(legend.position = "bottom",
        legend.title=element_blank(),
        legend.text=element_text(size=10))

taxa.leg = g_legend(leg_plot)

# Add axis labels
x.grob <- textGrob("Land-use", gp=gpar(fontface="bold", col="black", fontsize=14))
y.grob <- textGrob("Incorporator OTU count", gp=gpar(fontface="bold", col="black", fontsize=14), rot=90)
leg.grob <- textGrob("Phylum/Class", gp=gpar(col="black", fontsize=12))

# Put all plots together
taxa_count.plot = cowplot::plot_grid(sol.count.plot, insol.count.plot, nrow=1, rel_widths = c(0.7,1))
taxa_count.plot = arrangeGrob(taxa_count.plot, bottom = x.grob, left = y.grob)
taxa_count.plot = cowplot::plot_grid(taxa.leg, taxa_count.plot, nrow=2, rel_heights = c(0.4, 1))
taxa_count.plot = grid.arrange(arrangeGrob(taxa_count.plot, top = leg.grob))
taxa_count.plot
```

To make this a bit easier to understand, I'll group days based on whether they are early (i.e. before or around the day of peak mineralization for a given substrate) or late (i.e. after the day of peak mineralization for a given substrate).

```{r}
# Set timeperiods for each timepoint
timeperiods = data.frame(day = c(1, 3, 6, 14, 1, 3, 6, 14,
                                 6, 14, 30, 6, 14, 30, 6, 14, 30),
                         substrate = c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                       "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                       "13C-Van", "13C-Van", "13C-Van", 
                                       "13C-Cel", "13C-Cel", "13C-Cel",
                                       "13C-Pal", "13C-Pal", "13C-Pal"),
                         period = c("Early", "Early", "Late", "Late",
                                    "Early", "Early", "Late", "Late",
                                    "Early", "Late", "Late",
                                    "Early", "Early", "Late",
                                    "Early", "Early", "Late"))

# Get OTU counts for each taxa after grouping taxa with less than 5 labeled OTUs into a single group.
incorp_phyla.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  mutate(day = as.numeric(day)) %>%
  left_join(timeperiods, by = c("day", "substrate")) %>%
  mutate(taxa = ifelse(is.na(Phylum), "Unclassified bacteria",
                       ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum)))) %>%
  select(OTU, taxa, ecosystem, substrate, period) %>%
  unique

incorp_totalOTUcount.df = incorp_phyla.df %>%
  select(OTU, taxa) %>%
  unique() %>%
  group_by(taxa) %>%
  summarize(total_OTU_count = n()) %>%
  as.data.frame %>%
  arrange(total_OTU_count) %>%
  mutate(taxa2 = ifelse(total_OTU_count < 5, "OTU count < 5", taxa))

incorp_taxa_count.df = incorp_phyla.df %>%
  left_join(incorp_totalOTUcount.df, by = "taxa") %>%
  group_by(taxa2, ecosystem, substrate, period) %>%
  summarize(OTU_count = n()) %>%
  as.data.frame %>%
  mutate(eco = factor(ifelse(ecosystem == "agriculture", "A",
                             ifelse(ecosystem == "meadow", "M",
                                    ifelse(ecosystem == "forest", "F", NA))), 
                      levels = c("A", "M", "F")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))

```

Plot this as a stacked bar plot

```{r, fig.height=6.69291, fig.width=6.69291}

# Set colors for the taxa
source("/Users/sambarnett/Desktop/paul_tol_colors.R")
taxa = unique(incorp_taxa_count.df[incorp_taxa_count.df$taxa != "OTU count < 5",]$taxa2)
taxa.cols = c(paultol_colors(length(taxa)), "#777777")
names(taxa.cols) = c(sort(taxa), "OTU count < 5")

# Arrange taxa labels for nicer plotting
incorp_taxa_count.df$taxa2 = factor(incorp_taxa_count.df$taxa2,
                                    levels=c(names(taxa.cols)))

# Plot for soluble substrates (measured on days 1, 3, 6, and 14)
taxa_count_grouped.plot = ggplot(data=incorp_taxa_count.df, aes(x=eco, y=OTU_count, fill=taxa2,)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=taxa.cols) +
  labs(x = "Land-use", y="OTU count", fill="Phylum/Class") +
  guides(fill = guide_legend(ncol=1)) +
  theme_bw() +
  theme(legend.position = "right",
        legend.title=element_text(size=12),
        legend.text=element_text(size=10),
        axis.title = element_text(size=14),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_grid(substrate~period, scales="free_y")

taxa_count_grouped.plot

```

I also want at table with all this data, but this time without grouping the OTUs with less than 5 OTUs.

```{r}
incorp_phyla.df %>%
  select(OTU, taxa, ecosystem) %>%
  unique %>%
  group_by(taxa, ecosystem) %>%
  summarize(OTU_count = n()) %>%
  as.data.frame %>%
  tidyr::spread(key=ecosystem, value=OTU_count)

incorp_phyla.df %>%
  group_by(ecosystem, substrate, period) %>%
  summarize(OTU_count = n()) %>%
  as.data.frame %>%
  tidyr::spread(key=ecosystem, value=OTU_count)

incorp_phyla.df %>%
  group_by(taxa, ecosystem, substrate, period) %>%
  summarize(OTU_count = n()) %>%
  as.data.frame %>%
  tidyr::spread(key=ecosystem, value=OTU_count) %>%
  replace(is.na(.), 0) %>%
  mutate(substrate = factor(substrate, levels = c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal"))) %>%
  arrange(taxa, substrate, period)


```

### Phylum abundances

Now I want to see how abundant the incorporators are across the microcosms, grouped by their phylum.

```{r}
# Get metadata
metadata = data.frame(sample_data(unfrac.rare.physeq)) %>%
  rename(microcosm_substrate = substrate) %>%
  select(X.Sample, ecosystem, day, microcosm_substrate, microcosm_replicate)

# Get OTU relative abundances
OTU_table.df = data.frame(otu_table(transform_sample_counts(unfrac.rare.physeq, function(x) x / sum(x)))) %>%
  tibble::rownames_to_column(var="OTU") %>%
  tidyr::gather(key="X.Sample", value="abundance", -OTU) %>%
  mutate(X.Sample = gsub("13C.", "13C-", gsub("12C.", "12C-", gsub("H2O.", "H2O-", X.Sample))),
         abundance = abundance*100) %>%
  filter(abundance > 0) %>%
  left_join(metadata, by="X.Sample")

# Get incorporators and their pylum
incorp_phyla.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  mutate(taxa = ifelse(is.na(Phylum), "Unclassified bacteria",
                       ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum))),
         day = as.numeric(day)) %>%
  select(OTU, taxa, ecosystem, day, substrate)

# Group taxa that have less than 1% abundance in any sample
OTU_table.taxa.df = inner_join(OTU_table.df, incorp_phyla.df, by = c("OTU", "ecosystem", "day")) %>%
  group_by(taxa, X.Sample, substrate, ecosystem, day, microcosm_substrate, microcosm_replicate) %>%
  mutate(taxa_abd = sum(abundance)) %>%
  ungroup %>%
  group_by(taxa) %>%
  mutate(max_taxa_abd = max(taxa_abd)) %>%
  ungroup %>%
  mutate(taxa = ifelse(max_taxa_abd < 1, "Less than 1%", taxa)) %>%
  group_by(taxa, X.Sample, substrate, ecosystem, day, microcosm_substrate, microcosm_replicate) %>%
  summarize(taxa_abd = sum(abundance)) %>%
  as.data.frame %>%
  filter(microcosm_substrate != "H2O-Con") %>%
  mutate(ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")),
         microcosm_substrate = factor(microcosm_substrate, levels=c("12C-Con", "13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")),
         day_factor = factor(day, levels = c(0, 1, 3, 6, 14, 30)))

```

Now I'll plot these abundances.

```{r, fig.height=6.69291, fig.width=6.69291}
# Set colors for the taxa
source("/Users/sambarnett/Desktop/paul_tol_colors.R")
taxa = unique(OTU_table.taxa.df[OTU_table.taxa.df$taxa != "Less than 1%",]$taxa)
taxa.cols = c(paultol_colors(length(taxa)), "#777777")
names(taxa.cols) = c(sort(taxa), "Less than 1%")

# Get positions for each replicate in the plot. Separating bars by their day with separate bars per replicate
day.shift = data.frame(substrate = c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                     "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                     "13C-Van", "13C-Van", "13C-Van",
                                     "13C-Cel", "13C-Cel", "13C-Cel",
                                     "13C-Pal", "13C-Pal", "13C-Pal"),
                       day_factor = factor(c(1, 3, 6, 14, 1, 3, 6, 14,
                                      6, 14, 30, 6, 14, 30, 6, 14, 30)),
                       shift = c(0, 1, 2, 3, 0, 1, 2, 3, 
                                 0, 1, 2, 0, 1, 2, 0, 1, 2))

plot.positions = OTU_table.taxa.df %>%
  select(X.Sample, substrate, ecosystem, day_factor, microcosm_substrate, microcosm_replicate) %>%
  unique %>%
  arrange(day_factor, microcosm_substrate, microcosm_replicate) %>%
  group_by(ecosystem, substrate) %>%
  mutate(rank = row_number()) %>%
  ungroup %>%
  left_join(day.shift) %>%
  mutate(position = rank + shift) %>%
  select(X.Sample, ecosystem, substrate, position)

OTU_table.taxa.sum = left_join(OTU_table.taxa.df, plot.positions, by=c("X.Sample", "ecosystem", "substrate")) %>%
  mutate(taxa = factor(taxa, levels=names(taxa.cols)),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))


plot.xlab = OTU_table.taxa.sum %>%
  select(day_factor, position, ecosystem, substrate) %>%
  unique %>%
  group_by(day_factor, ecosystem, substrate) %>%
  summarize(midpoint = mean(position)) %>%
  as.data.frame

# Plot
ggplot(data=OTU_table.taxa.sum, aes(x=position, y=taxa_abd)) +
  geom_bar(stat="identity", aes(fill=taxa, color=taxa)) +
  geom_text(data=plot.xlab, aes(x=midpoint, label=day_factor), y=-7, size=4.5) +
  lims(y=c(-7, 100)) +
  labs(x="Time since substrate addition (days)", y="Relative abundance (% of reads)",
       fill="Phylum/Class", color="Phylum/Class") +
  scale_fill_manual(values=taxa.cols) +
  scale_color_manual(values=taxa.cols) +
  theme_bw() +
  theme(legend.position = "right",
        legend.title=element_text(size=12),
        legend.text=element_text(size=10),
        axis.title = element_text(size=14),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.ticks.x = element_blank(),
        strip.text = element_text(size=10)) +
  facet_grid(ecosystem~substrate, scales="free_x")

# Plot
ggplot(data=OTU_table.taxa.sum, aes(x=position, y=taxa_abd)) +
  geom_bar(stat="identity", aes(fill=taxa, color=taxa)) +
  geom_text(data=plot.xlab, aes(x=midpoint, label=day_factor), y=-10, size=4.5) +
  lims(y=c(-15, 100)) +
  labs(x="Time since substrate addition (days)", y="Relative abundance (% of reads)",
       fill="Phylum/Class", color="Phylum/Class") +
  scale_fill_manual(values=taxa.cols) +
  scale_color_manual(values=taxa.cols) +
  theme_bw() +
  theme(legend.position = "right",
        legend.title=element_text(size=12),
        legend.text=element_text(size=10),
        axis.title = element_text(size=14),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.ticks.x = element_blank(),
        strip.text = element_text(size=10)) +
  facet_grid(substrate~ecosystem)

```

### Specific groups

Now its time to look at a few specific groups of incorporators.

```{r}
# Get metadata
metadata = data.frame(sample_data(unfrac.rare.physeq)) %>%
  rename(microcosm_substrate = substrate) %>%
  select(X.Sample, ecosystem, day, microcosm_substrate, microcosm_replicate)

# Get incorporator OTU relative abundances in each microcosm
incorp_OTU_table.df = data.frame(otu_table(transform_sample_counts(unfrac.rare.physeq, function(x) x / sum(x)))) %>%
  tibble::rownames_to_column(var="OTU") %>%
  filter(OTU %in% unique(l2fc.df[l2fc.df$padj < 0.05,]$OTU)) %>%
  tidyr::gather(key="X.Sample", value="abundance", -OTU) %>%
  mutate(X.Sample = gsub("13C.", "13C-", gsub("12C.", "12C-", gsub("H2O.", "H2O-", X.Sample))),
         abundance = abundance*100) %>%
  left_join(metadata, by="X.Sample")
```

#### Top 5 most abundant OTUs
```{r}
incorp_status.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem, day)

incorp_class.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem, day, Domain, Phylum, Class, Order, Family, Genus, Species)

top5_incorp.df = incorp_OTU_table.df %>%
  group_by(OTU, ecosystem, day) %>%
  summarize(mean_abundance = mean(abundance),
            sd_abundance = sd(abundance),
            n_microcosms = n()) %>%
  as.data.frame %>%
  mutate(SE_abundance = sd_abundance/sqrt(n_microcosms)) %>%
  group_by(OTU, ecosystem) %>%
  mutate(max_abundance = max(mean_abundance)) %>%
  ungroup %>%
  group_by(ecosystem) %>%
  mutate(rank = dense_rank(desc(max_abundance))) %>%
  ungroup %>%
  arrange(rank) %>%
  filter(rank <= 5)
```



#### Deltaproteobacteria

Some deltaproteobacteria are known to be predators. Lets see if we see any signs of secondary feeding by these organisms. Secondary feeding might be indicated by late incorporation and/or increased abundance late in the experiment of these OTUs.

```{r, fig.height=6.69291, fig.width=6.69291}
# Get Deltaproteobacteria incorporators
delta_incorp.df = l2fc.df %>%
  filter(padj < 0.05, Class == "Deltaproteobacteria") %>%
  select(OTU, ecosystem, day, substrate, Phylum, Class, Order, Family, Genus, Species) %>%
  mutate(day = as.numeric(day))

# Get these incorporator taxonomies
delta_incorp.tax.df = delta_incorp.df %>%
  select(OTU, Phylum, Class, Order, Family, Genus, Species)
  
# Get these incorporator statuses
delta_incorp.status.df = delta_incorp.df %>%
  select(OTU, ecosystem, day, substrate) %>%
  mutate(status = "Labeled") %>%
  tidyr::spread(key="substrate", value = "status")

# Merge dataframes to get abundances and status of these incorporators
delta_incorp_abd.df = incorp_OTU_table.df %>%
  filter(microcosm_substrate != "H2O-Con") %>%
  select(OTU, X.Sample, abundance, ecosystem, day) %>%
  filter(OTU %in% unique(delta_incorp.df$OTU)) %>%
  full_join(delta_incorp.status.df, by = c("OTU", "ecosystem", "day")) %>%
  tidyr::gather(key="substrate", value="status", -OTU, -X.Sample, -abundance, -ecosystem, -day) %>%
  mutate(status = ifelse(is.na(status), "Unlabeled", "Labeled")) %>%
  left_join(delta_incorp.tax.df)
  
# Get the average abundance of each incorporator on each day and ecosystem
delta_incorp_abd.sum = delta_incorp_abd.df %>%
  group_by(OTU, ecosystem, day, substrate, status, Phylum, Class, Order, Family, Genus, Species) %>%
  summarize(mean_abd = mean(abundance),
            sd_abd = sd(abundance),
            n_rep = n()) %>%
  as.data.frame %>%
  mutate(SE_abd = sd_abd/sqrt(n_rep),
         status_num = ifelse(status == "Labeled", 1, 0)) %>%
  group_by(OTU, ecosystem, substrate) %>%
  mutate(max_status = max(status_num)) %>%
  ungroup %>%
  filter(max_status == 1) %>%
  mutate(ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))

# Plot
ggplot(data=delta_incorp_abd.sum, aes(x=day, y=mean_abd)) +
  geom_line(aes(color=OTU)) +
  geom_point(aes(color=OTU, shape=status)) +
  scale_shape_manual(values=c(Unlabeled = 1, Labeled = 16)) +
  labs(x="Time since substrate addition (days)", y="Mean relative abundance (% of reads)") +
  theme_bw() +
  theme(legend.position = "none",
        legend.title=element_text(size=12),
        legend.text=element_text(size=10),
        axis.title = element_text(size=14),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.ticks.x = element_blank(),
        strip.text = element_text(size=10)) +
  facet_grid(substrate ~ ecosystem)
```

What is the abundance of labeled Deltaproteobacteria over time?

```{r, fig.height=6.69291, fig.width=6.69291}
# Get Deltaproteobacteria incorporators
delta_incorp_status.df = l2fc.df %>%
  filter(padj < 0.05, Class == "Deltaproteobacteria") %>%
  select(OTU, ecosystem, day, substrate) %>%
  mutate(day = as.numeric(day),
         status = "Labeled") %>%
  tidyr::spread(key=substrate, value=status)
  

# Get all Deltaproteobacteria OTUs
delta_OTUs.df = data.frame(tax_table(unfrac.rare.physeq)) %>%
  tibble::rownames_to_column(var="OTU") %>%
  filter(Class == "Deltaproteobacteria")

# Get OTU relative abundances for Deltaproteobacteria
delta_abd.df = data.frame(otu_table(transform_sample_counts(unfrac.rare.physeq, function(x) x / sum(x)))) %>%
  tibble::rownames_to_column(var="OTU") %>%
  filter(OTU %in% delta_OTUs.df$OTU) %>%
  tidyr::gather(key="X.Sample", value="abundance", -OTU) %>%
  filter(abundance > 0) %>%
  mutate(X.Sample = gsub("13C.", "13C-", gsub("12C.", "12C-", gsub("H2O.", "H2O-", X.Sample))),
         abundance = abundance*100) %>%
  left_join(metadata, by="X.Sample")

# Get summary table of Deltaproteobacteria relative abundances and labeling
delta_abd_lab.sum = left_join(delta_abd.df, delta_incorp_status.df, by = c("OTU", "ecosystem", "day")) %>%
  tidyr::gather(key=substrate, value=status, -OTU, -X.Sample, -abundance, 
                -ecosystem, -day, -microcosm_substrate, -microcosm_replicate) %>%
  mutate(status = ifelse(is.na(status), "Unlabeled", status)) %>%
  group_by(X.Sample, ecosystem, day, microcosm_substrate, microcosm_replicate, substrate, status) %>%
  summarize(abundance = sum(abundance)) %>%
  as.data.frame

# Get positions for each replicate in the plot. Separating bars by their day with separate bars per replicate
day.shift = data.frame(substrate = c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                     "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                     "13C-Van", "13C-Van", "13C-Van",
                                     "13C-Cel", "13C-Cel", "13C-Cel",
                                     "13C-Pal", "13C-Pal", "13C-Pal"),
                       day = c(1, 3, 6, 14, 1, 3, 6, 14,
                               6, 14, 30, 6, 14, 30, 6, 14, 30),
                       shift = c(0, 1, 2, 3, 0, 1, 2, 3, 
                                 0, 1, 2, 0, 1, 2, 0, 1, 2))

plot.positions = delta_abd_lab.sum %>%
  select(X.Sample, substrate, ecosystem, day, microcosm_substrate, microcosm_replicate) %>%
  unique %>%
  inner_join(day.shift) %>%
  arrange(day, microcosm_substrate, microcosm_replicate) %>%
  group_by(ecosystem, substrate) %>%
  mutate(rank = row_number()) %>%
  ungroup %>%
  mutate(position = rank + shift) %>%
  select(X.Sample, ecosystem, substrate, position)

delta_abd_lab.sum = inner_join(delta_abd_lab.sum, plot.positions, by = c("X.Sample", "ecosystem", "substrate")) %>%
  mutate(ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")))

# Plot
ggplot(data=delta_abd_lab.sum, aes(x=position, y=abundance)) +
  geom_bar(stat="identity", aes(fill=status, color=status)) +
  geom_text(data=plot.xlab, aes(x=midpoint, label=day_factor), y=-0.75, size=4.5) +
  lims(y=c(-1.25, 6.5)) +
  labs(x="Time since substrate addition (days)", y="Relative abundance (% of reads)",
       fill="Status", color="Status") +
  scale_fill_manual(values=c(Labeled="red", Unlabeled="grey70")) +
  scale_color_manual(values=c(Labeled="red", Unlabeled="grey70")) +
  theme_bw() +
  theme(legend.position = "right",
        legend.title=element_text(size=12),
        legend.text=element_text(size=10),
        axis.title = element_text(size=14),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.ticks.x = element_blank(),
        strip.text = element_text(size=10)) +
  facet_grid(substrate~ecosystem, scales = "free")


```

How many labeled deltaproteobacteria are there in each sample?

```{r, fig.height=6.69291, fig.width=6.69291}
# Get Deltaproteobacteria incorporators
delta_incorp.df = l2fc.df %>%
  filter(padj < 0.05, Class == "Deltaproteobacteria") %>%
  select(OTU, ecosystem, day, substrate, Order) %>%
  mutate(taxa = ifelse(is.na(Order), "Unclassified", as.character(Order)),
         day = as.numeric(day))

# Get OTU counts for each order
delta_incorp.sum = delta_incorp.df %>%
  group_by(ecosystem, day, substrate, taxa) %>%
  summarize(count = n()) %>%
  as.data.frame %>%
  mutate(ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")),
         eco = factor(ifelse(ecosystem == "agriculture", "A",
                             ifelse(ecosystem == "meadow", "M",
                                    ifelse(ecosystem == "forest", "F", NA))), 
                      levels = c("A", "M", "F")))

# Set colors for the taxa
source("/Users/sambarnett/Desktop/paul_tol_colors.R")
taxa = unique(delta_incorp.sum[delta_incorp.sum$taxa != "Unclassified",]$taxa)
taxa.cols = c(paultol_colors(length(taxa)), "#777777")
names(taxa.cols) = c(sort(taxa), "Unclassified")

# Plot OTU counts
delta_DOC_count.plot = ggplot(data=delta_incorp.sum[delta_incorp.sum$substrate %in% c("13C-Xyl", "13C-Ami"),], 
                              aes(x=eco, y=count)) +
  geom_bar(stat="identity", aes(fill=taxa, color=taxa)) +
  scale_fill_manual(values=taxa.cols) +
  scale_color_manual(values=taxa.cols) +
  theme_bw() +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_grid(day~substrate)

delta_POC_count.plot = ggplot(data=delta_incorp.sum[delta_incorp.sum$substrate %in% c("13C-Van", "13C-Cel", "13C-Pal"),], 
                              aes(x=eco, y=count)) +
  geom_bar(stat="identity", aes(fill=taxa, color=taxa)) +
  scale_fill_manual(values=taxa.cols) +
  scale_color_manual(values=taxa.cols) +
  theme_bw() +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_grid(day~substrate)

delta_count_leg.plot = ggplot(data=delta_incorp.sum, aes(x=eco, y=count)) +
  geom_bar(stat="identity", aes(fill=taxa, color=taxa)) +
  scale_fill_manual(values=taxa.cols) +
  scale_color_manual(values=taxa.cols) +
  labs(color = "Order in Deltaproteobacteria", fill="Order in Deltaproteobacteria") +
  theme_bw() +
  guides(fill = guide_legend(ncol=2)) +
  theme(legend.position = "right",
        legend.title=element_text(size=12),
        legend.text=element_text(size=10),
        legend.title.align=0.5)
delta.leg = g_legend(delta_count_leg.plot)

# Add axis labels
x.grob <- textGrob("Land-use", gp=gpar(fontface="bold", col="black", fontsize=14))
y.grob <- textGrob("Incorporator OTU count", gp=gpar(fontface="bold", col="black", fontsize=14), rot=90)

# Put all plots together
delta_count.plot = cowplot::plot_grid(delta.leg, delta_POC_count.plot, ncol=1, rel_heights = c(0.3,1))
delta_count.plot = cowplot::plot_grid(delta_DOC_count.plot, delta_count.plot, nrow=1, rel_widths = c(0.7,1))
delta_count.plot = arrangeGrob(delta_count.plot, bottom = x.grob, left = y.grob)
grid.arrange(delta_count.plot)

```

## Assembly of incorporators

### Starting incorporator abundance in microcosms (Day 0)

One idea of why an incorporator is labeled in a given land use would be that it has been selected for or is already present in high enough quantities. Granted I'm sure this is much more complicated than this, but I'm curious if the incorporators are in higher abundance in the land use in which they are labeled than in those in which they aren't.

#### Reguardless of substrate or timepoint
First I'll just run this analysis without reguard for substrate or timepoint. In other words, I just care if the OTU is labeled in a given land-use at least one day with one substrate.

```{r}
# Get the incorporators and the ecosystems they are labeled in
incorp.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem) %>%
  unique %>%
  mutate(status = ecosystem) %>%
  tidyr::spread(key=ecosystem, value=status) %>%
  tidyr::unite(col="lab_ecos", agriculture, meadow, forest, sep="-") %>%
  mutate(lab_ecos = gsub("NA-", "", gsub("-NA", "", gsub("-NA-", "-", lab_ecos))),
         n_ecos = nchar(lab_ecos) - nchar(gsub("-", "", lab_ecos)) + 1)

# Get labeling status
lab_status.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem) %>%
  unique %>%
  mutate(status = "Labeled")
  
# Get metadata
metadata = data.frame(sample_data(unfrac.rare.physeq))
# Get the incorporator abundances
day0_abd.df = data.frame(otu_table(transform_sample_counts(subset_samples(unfrac.rare.physeq, day == 0), function(x) x / sum(x)))) %>%
  tibble::rownames_to_column(var="OTU") %>%
  filter(OTU %in% incorp.df$OTU) %>%
  tidyr::gather(key = "X.Sample", value="abundance", -OTU) %>%
  mutate(X.Sample = gsub("12C.Con", "12C-Con", X.Sample)) %>%
  left_join(metadata, by = "X.Sample") %>%
  group_by(OTU, ecosystem) %>%
  summarize(mean_abundance = mean(abundance)) %>%
  as.data.frame %>%
  group_by(OTU) %>%
  mutate(max_abundance = max(mean_abundance)) %>%
  ungroup %>%
  filter(max_abundance >= 0.001) %>%
  select(-max_abundance) %>%
  tidyr::spread(key=ecosystem, value=mean_abundance) %>%
  inner_join(incorp.df, by="OTU") %>%
  tidyr::gather(key="ecosystem", value="mean_abundance", -OTU, -lab_ecos, -n_ecos) %>%
  mutate(mean_abundance = ifelse(is.na(mean_abundance), 0, mean_abundance)) %>%
  full_join(lab_status.df, by=c("OTU", "ecosystem")) %>%
  mutate(status = ifelse(is.na(status), "Unlabeled", "Labeled"),
         ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         lab_ecos = factor(lab_ecos, levels=c("agriculture", "meadow", "forest",
                                              "agriculture-meadow", "agriculture-forest", "meadow-forest",
                                              "agriculture-meadow-forest")),
         OTU = factor(OTU),
         eco = factor(ifelse(ecosystem == "agriculture", "A",
                             ifelse(ecosystem == "meadow", "M",
                                    ifelse(ecosystem == "forest", "F", NA))), 
                      levels = c("A", "M", "F")))

# Filter out OTUs that are below the detection threshold in all three land use strategies
day0_detect.df = day0_abd.df %>%
  select(OTU, lab_ecos, ecosystem, mean_abundance) %>%
  tidyr::spread(key=ecosystem, value=mean_abundance) %>%
  mutate(sum_abd = agriculture+meadow+forest) %>%
  mutate(detect = ifelse(sum_abd > 0, "Detected", "Undetected")) %>%
  select(OTU, lab_ecos, detect)

day0_detect.lab = day0_detect.df %>%
  group_by(lab_ecos) %>%
  mutate(total_OTUs = n()) %>%
  ungroup %>%
  filter(detect == "Detected") %>%
  group_by(lab_ecos, total_OTUs) %>%
  summarize(det_OTUs = n()) %>%
  as.data.frame %>%
  #mutate(label = paste(lab_ecos, " (", paste(det_OTUs, total_OTUs, sep="/"), ")", sep="")) %>%
  mutate(label = paste(lab_ecos, total_OTUs, "OTUs", sep=" ")) %>%
  select(lab_ecos, label) %>%
  arrange(lab_ecos)
day0_detect.lab$label = factor(day0_detect.lab$label, 
                               levels = unique(day0_detect.lab$label))

day0_abd.df = filter(day0_abd.df, OTU %in% day0_detect.df[day0_detect.df$detect == "Detected",]$OTU) %>%
  left_join(day0_detect.lab, by="lab_ecos")

# Run the mixed effects model to compare abundance across land use
lme_res.df = data.frame()
for (labs in c("agriculture-meadow-forest", "agriculture-meadow", "agriculture-forest", 
               "meadow-forest", "agriculture", "meadow", "forest")){
  model = lme(mean_abundance ~ ecosystem, random = ~1|OTU,
              data=filter(day0_abd.df, lab_ecos == labs), method="REML")
  model.aov = anova(model)
  posthoc.res = lsmeans(model, pairwise~ecosystem, adjust="bonferroni")
  lme_res.df = rbind(lme_res.df, data.frame(posthoc.res$contrasts) %>%
                       mutate(lab_ecos = labs, lme_p.value = model.aov$`p-value`[2]))
}
lme_res.df
```


```{r, fig.height=6.69291, fig.width=6.69291}
# Get dataframe for significance bars
lme_sig.df = lme_res.df %>%
  filter(lme_p.value < 0.05) %>%
  mutate(xmin = ifelse(grepl("agriculture", contrast), 1.05, 2.05),
         xmax = ifelse(grepl("forest", contrast), 2.95, 1.95),
         ymin = ifelse(grepl("agriculture", contrast) & grepl("forest", contrast), 0.067, 0.062),
         ymax = ifelse(grepl("agriculture", contrast) & grepl("forest", contrast), 0.070, 0.065),
         xtext = ifelse(grepl("agriculture", contrast),
                        ifelse(grepl("meadow", contrast), 1.5, 2), 2.5),
         ytext = ifelse(grepl("agriculture", contrast) & grepl("forest", contrast), 0.071, 0.066),
         sig = ifelse(p.value < 0.001, "***",
                      ifelse(p.value < 0.01, "**",
                             ifelse(p.value < 0.05, "*", "NS"))),
         lab_ecos = factor(lab_ecos, levels=c("agriculture", "meadow", "forest",
                                              "agriculture-meadow", "agriculture-forest", "meadow-forest",
                                              "agriculture-meadow-forest"))) %>%
  left_join(day0_detect.lab, by="lab_ecos")

# How many incorporators were above the detection threshold for each group?
abv_det.df = day0_abd.df %>%
  mutate(detect = ifelse(mean_abundance > 0, 1, 0)) %>%
  group_by(label, ecosystem, status, eco) %>%
  summarize(n_OTUs_det = sum(detect)) %>%
  as.data.frame

ggplot(data=day0_abd.df, aes(x=eco, y=mean_abundance)) +
  geom_boxplot(aes(color=status), outlier.shape = 1) +
  geom_segment(data=lme_sig.df[lme_sig.df$p.value < 0.05,], 
               aes(x=xmin, xend=xmin, y=ymin, yend=ymax)) +
  geom_segment(data=lme_sig.df[lme_sig.df$p.value < 0.05,], 
               aes(x=xmin, xend=xmax, y=ymax, yend=ymax)) +
  geom_segment(data=lme_sig.df[lme_sig.df$p.value < 0.05,], 
               aes(x=xmax, xend=xmax, y=ymax, yend=ymin)) +
  geom_text(data=lme_sig.df[lme_sig.df$p.value < 0.05,],
            aes(label=sig, x=xtext, y=ytext), size=5) +
  geom_text(data=abv_det.df, aes(label=n_OTUs_det, x=eco), y=0.08, size=3) +
  ylim(c(0, 0.08)) +
  labs(x="Land-use", y="Mean relative abundance on day 0 (%)", color="Labeling status") +
  theme_bw() +
  theme(legend.position = c(0.70, 0.15),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12),
        axis.title = element_text(size=14),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_wrap(~label)
```

```{r}
length(unique(day0_abd.df$OTU))
length(unique(lab_status.df$OTU))
```

#### Within substrate

Now I want to see if we see the same trend for all substrates.

```{r}
# Get the incorporators and the ecosystems they are labeled in
incorp.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem, substrate) %>%
  unique %>%
  mutate(status = ecosystem) %>%
  tidyr::spread(key=ecosystem, value=status) %>%
  tidyr::unite(col="lab_ecos", agriculture, meadow, forest, sep="-") %>%
  mutate(lab_ecos = gsub("NA-", "", gsub("-NA", "", gsub("-NA-", "-", lab_ecos))),
         n_ecos = nchar(lab_ecos) - nchar(gsub("-", "", lab_ecos)) + 1)

# Get labeling status
lab_status.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  select(OTU, ecosystem, substrate) %>%
  unique %>%
  mutate(status = "Labeled")
  
# Get metadata
metadata = data.frame(sample_data(unfrac.rare.physeq))
# Get the incorporator abundances
day0_abd.df = data.frame(otu_table(transform_sample_counts(subset_samples(unfrac.rare.physeq, day == 0), function(x) x / sum(x)))) %>%
  tibble::rownames_to_column(var="OTU") %>%
  filter(OTU %in% incorp.df$OTU) %>%
  tidyr::gather(key = "X.Sample", value="abundance", -OTU) %>%
  mutate(X.Sample = gsub("12C.Con", "12C-Con", X.Sample)) %>%
  left_join(metadata, by = "X.Sample") %>%
  group_by(OTU, ecosystem) %>%
  summarize(mean_abundance = mean(abundance)) %>%
  as.data.frame %>%
  tidyr::spread(key=ecosystem, value=mean_abundance) %>%
  full_join(incorp.df, by="OTU") %>%
  tidyr::gather(key="ecosystem", value="mean_abundance", 
                -OTU, -lab_ecos, -n_ecos, -substrate) %>%
  mutate(mean_abundance = ifelse(is.na(mean_abundance), 0, mean_abundance)) %>%
  full_join(lab_status.df, by=c("OTU", "ecosystem", "substrate")) %>%
  mutate(status = ifelse(is.na(status), "Unlabeled", "Labeled"),
         ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         lab_ecos = factor(lab_ecos, levels=c("agriculture", "meadow", "forest",
                                              "agriculture-meadow", "agriculture-forest", "meadow-forest",
                                              "agriculture-meadow-forest")),
         OTU = factor(OTU),
         eco = factor(ifelse(ecosystem == "agriculture", "A",
                             ifelse(ecosystem == "meadow", "M",
                                    ifelse(ecosystem == "forest", "F", NA))), 
                      levels = c("A", "M", "F")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", 
                                                "13C-Cel", "13C-Pal")))

# Filter out OTUs that are below the detection threshold in all three land use strategies
day0_detect.df = day0_abd.df %>%
  select(OTU, lab_ecos, ecosystem, substrate, mean_abundance) %>%
  tidyr::spread(key=ecosystem, value=mean_abundance) %>%
  mutate(sum_abd = agriculture+meadow+forest) %>%
  mutate(detect = ifelse(sum_abd > 0, "Detected", "Undetected")) %>%
  select(OTU, substrate, lab_ecos, detect)

day0_detect.lab = day0_detect.df %>%
  group_by(lab_ecos, substrate) %>%
  mutate(total_OTUs = n()) %>%
  ungroup %>%
  filter(detect == "Detected") %>%
  group_by(lab_ecos, substrate, total_OTUs) %>%
  summarize(det_OTUs = n()) %>%
  as.data.frame %>%
  mutate(label = paste(lab_ecos, " (", 
                       paste(det_OTUs, total_OTUs, sep="/"), 
                       ")", sep="")) %>%
  select(lab_ecos, substrate, label) %>%
  arrange(lab_ecos)
day0_detect.lab$label = factor(day0_detect.lab$label, 
                               levels = unique(day0_detect.lab$label))

day0_abd.df = inner_join(day0_abd.df, day0_detect.df[day0_detect.df$detect == "Detected",], 
                         by=c("OTU", "lab_ecos", "substrate")) %>%
  left_join(day0_detect.lab, by=c("lab_ecos", "substrate"))

# Run the mixed effects model to compare abundance across land use
lme_res.df = data.frame()
for (sub in c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")){
  for (labs in c("agriculture-meadow-forest", "agriculture-meadow", "agriculture-forest", 
                 "meadow-forest", "agriculture", "meadow", "forest")){
    model = lme(mean_abundance ~ ecosystem, random = ~1|OTU,
                data=filter(day0_abd.df, lab_ecos == labs, substrate == sub),
                method="REML")
    model.aov = anova(model)
    posthoc.res = lsmeans(model, pairwise~ecosystem, adjust="bonferroni")
    lme_res.df = rbind(lme_res.df, data.frame(posthoc.res$contrasts) %>%
                         mutate(lab_ecos = labs, substrate = sub,
                                lme_p.value = model.aov$`p-value`[2]))
  }
}
lme_res.df
```

```{r, fig.height=6.69291, fig.width=6.69291}
# Get dataframe for significance bars
lme_sig.df = lme_res.df %>%
  filter(lme_p.value < 0.05) %>%
  mutate(xmin = ifelse(grepl("agriculture", contrast), 1.05, 2.05),
         xmax = ifelse(grepl("forest", contrast), 2.95, 1.95),
         ymin = ifelse(grepl("agriculture", contrast) & grepl("forest", contrast), 0.067, 0.062),
         ymax = ifelse(grepl("agriculture", contrast) & grepl("forest", contrast), 0.070, 0.065),
         xtext = ifelse(grepl("agriculture", contrast),
                        ifelse(grepl("meadow", contrast), 1.5, 2), 2.5),
         ytext = ifelse(grepl("agriculture", contrast) & grepl("forest", contrast), 0.071, 0.066),
         sig = ifelse(p.value < 0.001, "***",
                      ifelse(p.value < 0.01, "**",
                             ifelse(p.value < 0.05, "*", "NS"))),
         lab_ecos = factor(lab_ecos, levels=c("agriculture", "meadow", "forest",
                                              "agriculture-meadow", "agriculture-forest", "meadow-forest",
                                              "agriculture-meadow-forest"))) %>%
  left_join(day0_detect.lab, by=c("lab_ecos", "substrate"))

# How many incorporators were above the detection threshold for each group?
abv_det.df = day0_abd.df %>%
  mutate(detect = ifelse(mean_abundance > 0, 1, 0)) %>%
  group_by(label, ecosystem, substrate, status, eco) %>%
  summarize(n_OTUs_det = sum(detect)) %>%
  as.data.frame

plot.list = list()
for (sub in c("13C-Xyl", "13C-Ami", "13C-Van", "13C-Cel", "13C-Pal")){
plot.list[[sub]] = ggplot(data=day0_abd.df[day0_abd.df$substrate == sub,], 
                          aes(x=eco, y=mean_abundance)) +
    geom_boxplot(aes(color=status), outlier.shape = 1) +
    geom_segment(data=lme_sig.df[lme_sig.df$p.value < 0.05 & lme_sig.df$substrate == sub,], 
                 aes(x=xmin, xend=xmin, y=ymin, yend=ymax)) +
    geom_segment(data=lme_sig.df[lme_sig.df$p.value < 0.05 & lme_sig.df$substrate == sub,], 
                 aes(x=xmin, xend=xmax, y=ymax, yend=ymax)) +
    geom_segment(data=lme_sig.df[lme_sig.df$p.value < 0.05 & lme_sig.df$substrate == sub,], 
                 aes(x=xmax, xend=xmax, y=ymax, yend=ymin)) +
    geom_text(data=lme_sig.df[lme_sig.df$p.value < 0.05 & lme_sig.df$substrate == sub,],
              aes(label=sig, x=xtext, y=ytext), size=5) +
    geom_text(data=abv_det.df[abv_det.df$substrate == sub,], aes(label=n_OTUs_det, x=eco), y=0.08, size=3) +
    ylim(c(0, 0.08)) +
    labs(x="Land-use", y="Mean relative abundance on day 0 (%)", 
         color="Labeling status", title=sub) +
    theme_bw() +
    theme(legend.position = c(0.70, 0.15),
          legend.title=element_text(size=14),
          legend.text=element_text(size=12),
          axis.title = element_text(size=14),
          axis.text = element_text(size=12),
          strip.text = element_text(size=10)) +
    facet_wrap(~label)
}

plot.list
```


#### Initial labeling

Now I want to see if the starting abundance of an OTU correlates with the day in which it is first labeled. I expect early incorporators for each carbon source will have a higher starting abundance than those labeled later.

```{r}
# Get the incorporators and the ecosystems they are labeled in
incorp.df = l2fc.df %>%
  filter(padj < 0.05) %>%
  group_by(OTU, ecosystem, substrate) %>%
  summarize(init_day = min(as.numeric(day))) %>%
  as.data.frame

# Get the initial incorporator abundances
day0_abd.df = data.frame(otu_table(transform_sample_counts(subset_samples(unfrac.rare.physeq, day == 0), function(x) x / sum(x)))) %>%
  tibble::rownames_to_column(var="OTU") %>%
  filter(OTU %in% incorp.df$OTU) %>%
  tidyr::gather(key = "X.Sample", value="abundance", -OTU) %>%
  mutate(X.Sample = gsub("12C.Con", "12C-Con", X.Sample)) %>%
  left_join(metadata, by = "X.Sample") %>%
  group_by(OTU, ecosystem) %>%
  summarize(mean_abundance = mean(abundance)) %>%
  as.data.frame %>%
  tidyr::spread(key=ecosystem, value=mean_abundance) %>%
  full_join(unique(select(incorp.df, OTU)), by="OTU") %>%
  tidyr::gather(key="ecosystem", value="mean_abundance", -OTU) %>%
  mutate(mean_abundance = ifelse(is.na(mean_abundance), 0, mean_abundance)) %>%
  inner_join(incorp.df, by = c("OTU", "ecosystem")) %>%
  mutate(ecosystem = factor(ecosystem, levels=c("agriculture", "meadow", "forest")),
         substrate = factor(substrate, levels=c("13C-Xyl", "13C-Ami", "13C-Van", 
                                                "13C-Cel", "13C-Pal")))


ggplot(data=day0_abd.df, aes(x=as.factor(init_day), y=log10(mean_abundance+ 0.001))) +
  geom_boxplot() +
  labs(x="Day of first labeling", y="Mean relative abundance on day 0 (%)") +
  theme_bw() +
  theme(legend.title=element_text(size=14),
        legend.text=element_text(size=12),
        axis.title = element_text(size=14),
        axis.text = element_text(size=12),
        strip.text = element_text(size=10)) +
  facet_grid(ecosystem ~ substrate, scales="free_x")
```



## Figure for ASM microbe
```{r, fig.height=7, fig.width=7}
landuse.conv = data.frame(ecosystem = c("agriculture", "meadow", "forest"),
                          landuse = factor(c("Cropland", "Old-field", "Forest"), levels = c("Cropland", "Old-field", "Forest")))

# Dataframe for converting day to a time period (early and late)
subtime.df = data.frame(day = c(1, 3, 6, 14, 1, 3, 6, 14,
                                6, 14, 30, 6, 14, 30, 6, 14, 30),
                        substrate = factor(c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                             "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                             "13C-Van", "13C-Van", "13C-Van", 
                                             "13C-Cel", "13C-Cel", "13C-Cel",
                                             "13C-Pal", "13C-Pal", "13C-Pal"), 
                                           levels = c("13C-Xyl", "13C-Ami", "13C-Van",
                                                      "13C-Cel", "13C-Pal")),
                        period = c("Early", "Early", "Late", "Late",
                                   "Early", "Early", "Late", "Late",
                                   "Early", "Late", "Late",
                                   "Early", "Early", "Late",
                                   "Early", "Early", "Late"))
landuse.col = c("Cropland"="#00BA38", "Old-field"="#619CFF", "Forest"="#F8766D")

early.df = l2fc.df %>% 
  mutate(day = as.numeric(as.character(day))) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(padj < 0.05, period == "Early") %>%
  mutate(taxa = ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum))) %>%
  mutate(taxa = ifelse(is.na(taxa), "Unclassified", taxa)) %>%
  select(taxa, OTU, substrate, ecosystem) %>%
  unique %>%
  group_by(substrate, ecosystem, taxa) %>%
  summarize(n_OTU = n()) %>%
  left_join(landuse.conv, by="ecosystem") %>%
  mutate(Diversity = NA) %>%
  as.data.frame

## Get incorporator counts and phylogenetic diversity
early.mat = l2fc.df %>% 
  mutate(day = as.numeric(as.character(day))) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(padj < 0.05, period == "Early") %>%
  mutate(eco_sub = paste(ecosystem, substrate, sep="_"), 
         labeled = 1) %>%
  select(OTU, eco_sub, labeled) %>%
  unique %>%
  tidyr::spread(key=eco_sub, value=labeled) %>%
  tibble::column_to_rownames(var="OTU")
early.mat[is.na(early.mat)] = 0

tips2drop = tree$tip.label[!(tree$tip.label %in% unique(l2fc.df$OTU))]
SIP.tree = ape::drop.tip(phy = tree, tip = tips2drop)
early.total.df = picante::pd(t(early.mat), SIP.tree) %>%
  as.data.frame %>%
  tibble::rownames_to_column(var="eco_sub") %>%
  tidyr::gather(key="taxa", value="Diversity", -eco_sub) %>%
  tidyr::separate(eco_sub, into=c("ecosystem", "substrate"), sep="_") %>%
  left_join(landuse.conv, by="ecosystem") %>%
  mutate(taxa = ifelse(taxa == "PD", "Phylogenetic diversity",
                       ifelse(taxa == "SR", "Total incorporators", NA)),
         n_OTU = NA)

early.df = rbind(early.df, early.total.df)

taxa.list = c("Phylogenetic diversity", "Total incorporators", sort(unique(early.df$taxa[!(early.df$taxa %in% c("Total incorporators", "Phylogenetic diversity"))]), decreasing = TRUE))

early.df$taxa = factor(early.df$taxa, levels = taxa.list)

## Scale factor for the phylogenetic diversity sizes
scale.factor = max(filter(early.df, taxa=="Total incorporators")$Diversity)/max(filter(early.df, taxa=="Phylogenetic diversity")$Diversity)

## Font for the y axis, make richness and PD bold
vec_fontface <- ifelse(levels(early.df$taxa) %in% c("Phylogenetic diversity", "Total incorporators"),"bold","plain")


early.plot = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_tile(data=early.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  geom_point(data=filter(early.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  geom_point(data=filter(early.df, taxa=="Phylogenetic diversity"), aes(size = Diversity*scale.factor), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.y = element_text(face=vec_fontface),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)

```

```{r}
leg.plot1 = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_tile(data=early.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)

leg.plot2 = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_point(data=filter(early.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)

leg.plot3 = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_point(data=filter(early.df, taxa=="Phylogenetic diversity"), aes(size = Diversity), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Phylogenetic\ndiversity") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)



leg.plot1 = g_legend(leg.plot1)
leg.plot2 = g_legend(leg.plot2)
leg.plot3 = g_legend(leg.plot3)

```

```{r, fig.height=10, fig.width=7}
leg.plot = cowplot::plot_grid(leg.plot1, leg.plot2, leg.plot3, ncol=1, rel_heights = c(1,0.4, 0.4))
early.full.plot = cowplot::plot_grid(early.plot + theme(legend.position = "none"), leg.plot, ncol=2,
                                     rel_widths = c(1,0.3))

early_incorp_pub.plot = cowplot::plot_grid(venn_eco.plot, early.full.plot, ncol=1, rel_heights = c(0.4, 1),
                                           labels=c("A", "B"))
early_incorp_pub.plot

ggsave(early_incorp_pub.plot, filename = "/Users/sambarnett/Documents/Dissertation/figures/fig2_3.tiff", 
       device = "tiff", width = 7, height = 10, units = "in")
```


Late responders

```{r, fig.height=7, fig.width=7}
landuse.conv = data.frame(ecosystem = c("agriculture", "meadow", "forest"),
                          landuse = factor(c("Cropland", "Old-field", "Forest"), levels = c("Cropland", "Old-field", "Forest")))

# Dataframe for converting day to a time period (early and late)
subtime.df = data.frame(day = c(1, 3, 6, 14, 1, 3, 6, 14,
                                6, 14, 30, 6, 14, 30, 6, 14, 30),
                        substrate = factor(c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                             "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                             "13C-Van", "13C-Van", "13C-Van", 
                                             "13C-Cel", "13C-Cel", "13C-Cel",
                                             "13C-Pal", "13C-Pal", "13C-Pal"), 
                                           levels = c("13C-Xyl", "13C-Ami", "13C-Van",
                                                      "13C-Cel", "13C-Pal")),
                        period = c("Early", "Early", "Late", "Late",
                                   "Early", "Early", "Late", "Late",
                                   "Early", "Late", "Late",
                                   "Early", "Early", "Late",
                                   "Early", "Early", "Late"))
landuse.col = c("Cropland"="#00BA38", "Old-field"="#619CFF", "Forest"="#F8766D")

late.df = l2fc.df %>% 
  mutate(day = as.numeric(as.character(day))) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(padj < 0.05, period == "Late") %>%
  mutate(taxa = ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum))) %>%
  mutate(taxa = ifelse(is.na(taxa), "Unclassified", taxa)) %>%
  select(taxa, OTU, substrate, ecosystem) %>%
  unique %>%
  group_by(substrate, ecosystem, taxa) %>%
  summarize(n_OTU = n()) %>%
  left_join(landuse.conv, by="ecosystem") %>%
  mutate(Diversity = NA) %>%
  as.data.frame

## Get incorporator counts and phylogenetic diversity
late.mat = l2fc.df %>% 
  mutate(day = as.numeric(as.character(day))) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(padj < 0.05, period == "Late") %>%
  mutate(eco_sub = paste(ecosystem, substrate, sep="_"), 
         labeled = 1) %>%
  select(OTU, eco_sub, labeled) %>%
  unique %>%
  tidyr::spread(key=eco_sub, value=labeled) %>%
  tibble::column_to_rownames(var="OTU")
late.mat[is.na(late.mat)] = 0

tips2drop = tree$tip.label[!(tree$tip.label %in% unique(l2fc.df$OTU))]
SIP.tree = ape::drop.tip(phy = tree, tip = tips2drop)
late.total.df = picante::pd(t(late.mat), SIP.tree) %>%
  as.data.frame %>%
  tibble::rownames_to_column(var="eco_sub") %>%
  tidyr::gather(key="taxa", value="Diversity", -eco_sub) %>%
  tidyr::separate(eco_sub, into=c("ecosystem", "substrate"), sep="_") %>%
  left_join(landuse.conv, by="ecosystem") %>%
  mutate(taxa = ifelse(taxa == "PD", "Phylogenetic diversity",
                       ifelse(taxa == "SR", "Total incorporators", NA)),
         n_OTU = NA)

late.df = rbind(late.df, late.total.df)

taxa.list = c("Phylogenetic diversity", "Total incorporators", sort(unique(late.df$taxa[!(late.df$taxa %in% c("Total incorporators", "Phylogenetic diversity"))]), decreasing = TRUE))

late.df$taxa = factor(late.df$taxa, levels = taxa.list)

## Scale factor for the phylogenetic diversity sizes
scale.factor = max(filter(late.df, taxa=="Total incorporators")$Diversity)/max(filter(late.df, taxa=="Phylogenetic diversity")$Diversity)

## Font for the y axis, make richness and PD bold
vec_fontface <- ifelse(levels(late.df$taxa) %in% c("Phylogenetic diversity", "Total incorporators"),"bold","plain")


late.plot = ggplot(data=late.df, aes(x=substrate, y=taxa)) +
  geom_tile(data=late.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  geom_point(data=filter(late.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  geom_point(data=filter(late.df, taxa=="Phylogenetic diversity"), aes(size = Diversity*scale.factor), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.y = element_text(face=vec_fontface),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)

```

```{r}
leg.plot1 = ggplot(data=late.df, aes(x=substrate, y=taxa)) +
  geom_tile(data=late.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)

leg.plot2 = ggplot(data=late.df, aes(x=substrate, y=taxa)) +
  geom_point(data=filter(late.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)

leg.plot3 = ggplot(data=late.df, aes(x=substrate, y=taxa)) +
  geom_point(data=filter(late.df, taxa=="Phylogenetic diversity"), aes(size = Diversity), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_x_discrete(position = "top") +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Phylogenetic\ndiversity") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.text.x = element_text(angle=90, hjust=0),
        axis.text.x.top = element_text(vjust = 0.5),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        strip.placement = 'outside',
        strip.text = element_text(size=12),
        legend.text = element_text(size=12),
        legend.title = element_text(size=14)) +
  facet_wrap(~landuse, nrow=1)



leg.plot1 = g_legend(leg.plot1)
leg.plot2 = g_legend(leg.plot2)
leg.plot3 = g_legend(leg.plot3)

```

```{r, fig.height=7, fig.width=7}
late.plot

leg.plot = cowplot::plot_grid(leg.plot1, leg.plot2, leg.plot3, ncol=1, rel_heights = c(1,0.4, 0.4))
late_incorp_pub.plot = cowplot::plot_grid(late.plot + theme(legend.position = "none"), leg.plot, ncol=2,
                                          rel_widths = c(1,0.3))
late_incorp_pub.plot

ggsave(late_incorp_pub.plot, filename = "/Users/sambarnett/Documents/Dissertation/figures/figS2_9.tiff", 
       device = "tiff", width = 7, height = 7, units = "in")
```


## Figure for Pub
```{r, fig.height=4, fig.width=3.46457}
landuse.conv = data.frame(ecosystem = c("agriculture", "meadow", "forest"),
                          landuse = factor(c("Cropland", "Old-field", "Forest"), levels = c("Cropland", "Old-field", "Forest")))

# Dataframe for converting day to a time period (early and late)
subtime.df = data.frame(day = c(1, 3, 6, 14, 1, 3, 6, 14,
                                6, 14, 30, 6, 14, 30, 6, 14, 30),
                        substrate = factor(c("13C-Xyl", "13C-Xyl", "13C-Xyl", "13C-Xyl",
                                             "13C-Ami", "13C-Ami", "13C-Ami", "13C-Ami",
                                             "13C-Van", "13C-Van", "13C-Van", 
                                             "13C-Cel", "13C-Cel", "13C-Cel",
                                             "13C-Pal", "13C-Pal", "13C-Pal"), 
                                           levels = c("13C-Xyl", "13C-Ami", "13C-Van",
                                                      "13C-Cel", "13C-Pal")),
                        period = c("Early", "Early", "Late", "Late",
                                   "Early", "Early", "Late", "Late",
                                   "Early", "Late", "Late",
                                   "Early", "Early", "Late",
                                   "Early", "Early", "Late"))
landuse.col = c("Cropland"="#00BA38", "Old-field"="#619CFF", "Forest"="#F8766D")

early.df = l2fc.df %>% 
  mutate(day = as.numeric(as.character(day))) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(padj < 0.05, period == "Early") %>%
  mutate(taxa = ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum))) %>%
  mutate(taxa = ifelse(is.na(taxa), "Unclassified", taxa)) %>%
  select(taxa, OTU, substrate, ecosystem) %>%
  unique %>%
  group_by(substrate, ecosystem, taxa) %>%
  summarize(n_OTU = n()) %>%
  left_join(landuse.conv, by="ecosystem") %>%
  mutate(Diversity = NA) %>%
  as.data.frame

## Get incorporator counts and phylogenetic diversity
early.mat = l2fc.df %>% 
  mutate(day = as.numeric(as.character(day))) %>%
  left_join(subtime.df, by=c("day", "substrate")) %>%
  filter(padj < 0.05, period == "Early") %>%
  mutate(eco_sub = paste(ecosystem, substrate, sep="_"), 
         labeled = 1) %>%
  select(OTU, eco_sub, labeled) %>%
  unique %>%
  tidyr::spread(key=eco_sub, value=labeled) %>%
  tibble::column_to_rownames(var="OTU")
early.mat[is.na(early.mat)] = 0

tips2drop = tree$tip.label[!(tree$tip.label %in% unique(l2fc.df$OTU))]
SIP.tree = ape::drop.tip(phy = tree, tip = tips2drop)
early.total.df = picante::pd(t(early.mat), SIP.tree) %>%
  as.data.frame %>%
  tibble::rownames_to_column(var="eco_sub") %>%
  tidyr::gather(key="taxa", value="Diversity", -eco_sub) %>%
  tidyr::separate(eco_sub, into=c("ecosystem", "substrate"), sep="_") %>%
  left_join(landuse.conv, by="ecosystem") %>%
  mutate(taxa = ifelse(taxa == "PD", "Phylogenetic diversity",
                       ifelse(taxa == "SR", "Total incorporators", NA)),
         n_OTU = NA)

early.df = rbind(early.df, early.total.df)

taxa.list = c("Phylogenetic diversity", "Total incorporators", sort(unique(early.df$taxa[!(early.df$taxa %in% c("Total incorporators", "Phylogenetic diversity"))]), decreasing = TRUE))

early.df$taxa = factor(early.df$taxa, levels = taxa.list)

## Scale factor for the phylogenetic diversity sizes
scale.factor = max(filter(early.df, taxa=="Total incorporators")$Diversity)/max(filter(early.df, taxa=="Phylogenetic diversity")$Diversity)

## Font for the y axis, make richness and PD bold
vec_fontface <- ifelse(levels(early.df$taxa) %in% c("Phylogenetic diversity", "Total incorporators"),"bold","plain")


early.plot = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_tile(data=early.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  geom_point(data=filter(early.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  geom_point(data=filter(early.df, taxa=="Phylogenetic diversity"), aes(size = Diversity*scale.factor), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0) +
  scale_size_continuous(range = c(0, 2)) +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators", title = "Early 13C-Incorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.ticks = element_line(size=0.2), 
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.text.y = element_text(face=vec_fontface),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7),
        strip.placement = 'outside',
        strip.text = element_text(size=6),
        legend.text = element_text(size=6),
        legend.title = element_text(size=7, hjust=0.5),
        plot.title = element_text(size=7, hjust=0.5)) +
  facet_wrap(~landuse, nrow=1)


early.plot = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_tile(data=early.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  geom_point(data=filter(early.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  geom_point(data=filter(early.df, taxa=="Phylogenetic diversity"), aes(size = Diversity*scale.factor), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0, range = c(0, 1), limits=c(0,90), breaks=c(0, 20, 40, 60, 80)) +
  scale_size_continuous(range = c(0, 2)) +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators", 
       title = expression(bold("Early "^{13}*"C Incorporators"))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.ticks = element_line(size=0.2), 
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.text.y = element_text(face=vec_fontface),
        axis.text = element_text(size=6),
        axis.title.x = element_text(size=7),
        axis.title.y = element_text(size=7, margin=margin(r=-9)),
        strip.placement = 'outside',
        strip.text = element_text(size=6),
        legend.text = element_text(size=6),
        legend.title = element_text(size=7, hjust=0.5),
        plot.title = element_text(size=7, hjust=0.5, face = "bold")) +
  facet_wrap(~landuse, nrow=1)

early.plot

```

```{r}
leg.plot1 = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  #geom_tile(data=early.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  geom_tile(data=early.df, color="grey", aes(fill=landuse)) +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0, range = c(0, 1), limits=c(0,90), breaks=c(0, 20, 40, 60, 80)) +
  scale_size_continuous(range = c(0, 2)) +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.ticks = element_line(size=0.2), 
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.text.y = element_text(face=vec_fontface),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7),
        strip.placement = 'outside',
        strip.text = element_text(size=6),
        legend.text = element_text(size=6),
        legend.title = element_text(size=7, hjust=0.5),
        plot.title = element_text(size=7, hjust=0.5)) +
  facet_wrap(~landuse, nrow=1)

leg.plot2 = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_point(data=filter(early.df, taxa=="Total incorporators"), aes(size = Diversity), color="red") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0, range = c(0, 1), limits=c(0,90), breaks=c(0, 20, 40, 60, 80)) +
  scale_size_continuous(range = c(0, 2)) +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Total number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.ticks = element_line(size=0.2), 
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.text.y = element_text(face=vec_fontface),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7),
        strip.placement = 'outside',
        strip.text = element_text(size=6),
        legend.text = element_text(size=6),
        legend.title = element_text(size=7, hjust=0.5),
        plot.title = element_text(size=7, hjust=0.5)) +
  facet_wrap(~landuse, nrow=1)

leg.plot3 = ggplot(data=early.df, aes(x=substrate, y=taxa)) +
  geom_point(data=filter(early.df, taxa=="Phylogenetic diversity"), aes(size = Diversity), color="purple") +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0, range = c(0, 1), limits=c(0,90), breaks=c(0, 20, 40, 60, 80)) +
  scale_size_continuous(range = c(0, 2)) +
  labs(x="Substrate and land-use\n", y="Phylum/Class\n", 
       fill = "Land-use", alpha="Number of\nincorporators",
       size = "Phylogenetic\ndiversity") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.ticks = element_line(size=0.2), 
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.text.y = element_text(face=vec_fontface),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7),
        strip.placement = 'outside',
        strip.text = element_text(size=6),
        legend.text = element_text(size=6),
        legend.title = element_text(size=7, hjust=0.5),
        plot.title = element_text(size=7, hjust=0.5)) +
  facet_wrap(~landuse, nrow=1)



leg.plot1 = g_legend(leg.plot1)
leg.plot2 = g_legend(leg.plot2)
leg.plot3 = g_legend(leg.plot3)
```

```{r, fig.height=1, fig.width=0.5}
tile_leg.df = data.frame(landuse = factor(c("Cropland", "Cropland", "Cropland", "Cropland", "Cropland",
                                            "Old-field", "Old-field", "Old-field", "Old-field", "Old-field",
                                            "Forest", "Forest", "Forest", "Forest", "Forest"),
                                          levels=c("Cropland", "Old-field", "Forest")),
                         n_OTU = c(0, 20, 40, 60, 80, 0, 20, 40, 60, 80, 0, 20, 40, 60, 80),
                         alpha_labs = factor(c(0, 20, 40, 60, 80, 0, 20, 40, 60, 80, 0, 20, 40, 60, 80), 
                                             levels=c(80, 60, 40, 20, 0)))

leg.plot4 = ggplot(data=tile_leg.df, aes(x=landuse, y=alpha_labs)) +
  geom_tile(data=tile_leg.df, color="grey", aes(alpha=n_OTU, fill=landuse)) +
  scale_fill_manual(values = landuse.col) +
  scale_alpha_continuous(na.value=0, range = c(0, 1), limits=c(0,90), breaks=c(0, 20, 40, 60, 80)) +
  scale_size_continuous(range = c(0, 2)) +
  scale_y_discrete(position = "right") +
  labs(title="Number of\nincorporators") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        axis.ticks.x = element_blank(), 
        axis.ticks.y = element_line(size=0.2), 
        axis.text.x = element_blank(),
        axis.text.y = element_text(face=vec_fontface),
        axis.text = element_text(size=6),
        axis.title = element_blank(),
        legend.position = "none",
        plot.title = element_text(size=7, hjust=0.5))
leg.plot4
```


```{r, fig.height=5, fig.width=3.46457}
venn_eco.plot = plot(euler(venn_eco.df, by = list(test)), 
                     legend = FALSE, labels = FALSE,
                     #legend = list(labels = c("Cropland", "Old-field", "Forest"), fontsize=5),
                     quantities = list(TRUE, fontsize=5), strip=FALSE, fills=eco.col)
# Add axis labels
x.grob <- textGrob("Detection                                 Labeling", gp=gpar(fontface="bold", col="black", fontsize=7))
venn_eco.plot = grid.arrange(arrangeGrob(venn_eco.plot, top = x.grob))
venn_eco.plot


leg.plot = cowplot::plot_grid(leg.plot1, leg.plot4, leg.plot2, leg.plot3, ncol=1, rel_heights = c(1,1,1,1))

early_incorp_pub.plot = cowplot::plot_grid(venn_eco.plot, early.plot + theme(legend.position = "none"), ncol=1, rel_heights = c(0.4, 1),
                                           labels=c("a", "b"), label_size = 10)

early_incorp_leg_pub.plot = cowplot::plot_grid(early_incorp_pub.plot, leg.plot, ncol=2,
                                     rel_widths = c(1,0.3))

early_incorp_leg_pub.plot

ggsave(early_incorp_leg_pub.plot, filename = "/Users/sambarnett/Documents/Buckley Lab/FullCyc2/manuscript/Figures/Fig4.tiff", 
       device = "tiff", width = 3.46457, height = 5, units = "in")
```


